File: 2347. Best Poker Hand/Solution.cpp
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
    string bestHand(vector<int>& ranks, vector<char>& suits)
    {
        if (unordered_set<char>(suits.begin(), suits.end()).size() == 1)
            return "Flush";

        unordered_map<int, int> cnt;
        for (const int& x : ranks)
            cnt[x]++;
        if (cnt.size() == 5)
            return "High Card";
        for (const auto& [_, val] : cnt)
            if (val > 2)
                return "Three of a Kind";
        return "Pair";
    }
};

File: 2347. Best Poker Hand/Solution.py
from collections import Counter
from typing import List


class Solution:
    def bestHand(self, ranks: List[int], suits: List[str]) -> str:
        if max(suits) == min(suits):
            return "Flush"
        c = Counter(ranks)
        if len(c) == 5:
            return "High Card"
        for v in c.values():
            if v > 2:
                return "Three of a Kind"
        return "Pair"
